src/ctapipe/monitoring/aggregator.py

```diff
@@ -1,163 +1,402 @@
-is a monitoring table with columns describing the start and stop time of the chunk
-and the aggregated statistic values.
+These classes take as input an events table containing any event-wise quantities
+(e.g., images, scalars, vectors), divide it into time chunks, which may optionally
+overlap, and compute various aggregated statistics for each chunk. The statistics
+include the count, mean, median, and standard deviation. The result is a monitoring
+table with columns describing the start and stop time of the chunk and the
+aggregated statistic values.
+
+The aggregation is always performed along axis=0 (the event dimension), making
+these classes suitable for any N-dimensional event-wise data.
 """
 
 __all__ = [
+    "BaseChunking",
+    "SizeChunking",
+    "TimeChunking",
+    "BaseAggregator",
     "StatisticsAggregator",
     "PlainAggregator",
     "SigmaClippingAggregator",
 ]
 
-from abc import abstractmethod
+from abc import ABC, abstractmethod
 from collections import defaultdict
+from collections.abc import Generator
 
+import astropy.units as u
 import numpy as np
 from astropy.stats import sigma_clip
 from astropy.table import Table
 
 from ..containers import ChunkStatisticsContainer
-from ..core import TelescopeComponent
-from ..core.traits import Int
-
-__all__ = [
-    "StatisticsAggregator",
-    "PlainAggregator",
-    "SigmaClippingAggregator",
-]
+from ..core import Component
+from ..core.traits import AstroQuantity, ComponentName, Int
 
 
-class StatisticsAggregator(TelescopeComponent):
+class BaseChunking(Component, ABC):
     """
-    Base component to handle the computation of aggregated statistic values from a table
-    containing e.g. charges, peak times and/or charge variances (images).
+    Abstract base class for chunking strategies.
+
+    Chunking components divide tables into overlapping or non-overlapping chunks
+    for processing by aggregators.
     """
 
+    def __call__(self, table) -> Generator[Table, None, None]:
+        """
+        Generate chunks from the input table.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            Input table with 'time' and 'event_id' columns
+
+        Yields
+        ------
+        astropy.table.Table
+            Chunks of the input table
+        """
+        self._validate_table(table)
+        yield from self._generate_chunks(table)
+
+    def _validate_table(self, table):
+        """Validate that table has required columns."""
+        required_cols = ["time", "event_id"]
+        missing = [col for col in required_cols if col not in table.colnames]
+        if missing:
+            raise ValueError(f"Table must have columns: {missing}")
+
+    @abstractmethod
+    def _generate_chunks(self, table) -> Generator[Table, None, None]:
+        """Generate chunks from table. Implemented by subclasses."""
+        pass
+
+
+class SizeChunking(BaseChunking):
+    """Divides tables into chunks based on number of events."""
```

Comment:
Do we only plan to use this with events? Or might it also be used for a monitoring table of something other than events?

```suggestion
    """Divides tables into chunks based on number of samples."""
```

Reply:
Why does it need the event_id column? It just splits by len, right?